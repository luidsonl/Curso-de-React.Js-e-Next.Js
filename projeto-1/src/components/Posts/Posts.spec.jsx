import { render, screen } from '@testing-library/react'
import { Posts } from '.'

const props = {
    posts: [
        {   
            key: 1,
            id: 1,
            title: 'Title 1',
            body: 'Body 1',
            cover: 'img/cover1.png',
        },
        {
            key: 2,
            id: 2,
            title: 'Title 2',
            body: 'Body 2',
            cover: 'img/cover2.png',
        },
        {
            key: 3,
            id: 3,
            title: 'Title 3',
            body: 'Body 3',
            cover: 'img/cover3.png',
        },
    ]
}

describe('<Posts/>', ()=>{
    it('Should render posts correctly',()=>{
        render(<Posts {...props}/>)

        expect(screen.getAllByRole('heading', {name: /title/i}))
        .toHaveLength(3)

        expect(screen.getAllByRole('img', {name: /title/i}))
        .toHaveLength(3)

        expect(screen.getAllByText(/body/i))
        .toHaveLength(3)

        expect(screen.getByRole('img', {name: /title 3/i}))
        .toHaveAttribute('src', 'img/cover3.png')
    })

    it('Should not render posts if there r no posts', ()=>{
        render(<Posts/>)

        expect(screen.queryByRole('heading', {name: /title/i}))
        .not.toBeInTheDocument()
    })

    it('Should match snapshot', ()=>{
        const {container} = render(<Posts {...props}/>)

        expect(container.firstChild).toMatchSnapshot()
    })
})