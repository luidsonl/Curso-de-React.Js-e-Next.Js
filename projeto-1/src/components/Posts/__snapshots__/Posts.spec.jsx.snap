// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts/> Should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post-card"
  >
    <img
      alt="Title 1"
      src="img/cover1.png"
    />
    <div
      class="post-content"
    >
      <h1>
        Title 1
         
        1
      </h1>
      <p>
        Body 1
      </p>
    </div>
  </div>
  <div
    class="post-card"
  >
    <img
      alt="Title 2"
      src="img/cover2.png"
    />
    <div
      class="post-content"
    >
      <h1>
        Title 2
         
        2
      </h1>
      <p>
        Body 2
      </p>
    </div>
  </div>
  <div
    class="post-card"
  >
    <img
      alt="Title 3"
      src="img/cover3.png"
    />
    <div
      class="post-content"
    >
      <h1>
        Title 3
         
        3
      </h1>
      <p>
        Body 3
      </p>
    </div>
  </div>
</div>
`;
